src/main/java/com/example/purerestapi/repository/UserRepository.java

```diff
@@ -0,0 +1,37 @@
+package com.example.purerestapi.repository;
+
+import com.example.purerestapi.entity.User;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserRepository {
+
+  private final static Map<String, User> users = new ConcurrentHashMap<>();
+
+
+  public List<User> getAll() {
+    return users.values().stream().toList();
+  }
+  public void addUser(User user) {
+    if (user.getId() == null) {
+      user.setId(UUID.randomUUID().toString());
+    }
+    if (!users.containsKey(user.getId())) {
```

Comment:
It's an interesting part of the code in concept "what if there is already UUID in the map?". Do you expect duplicates?
If so, then better to make it in loop until unique UUID is found.